sessions/backend-update-delete/demo-end/components/Button/index.js

```diff
@@ -0,0 +1,20 @@
+import styled from "styled-components";
+
+const StyledButton = styled.button`
+  width: ${({ width }) => width};
+  margin: ${({ margin }) => margin};
+  appearance: none;
+  border: none;
+  background: var(--color-water-10);
+  font-size: larger;
+  padding: 0.5rem 1rem;
+  box-shadow: 0px 1px 5px -2px var(--color-granite);
+`;
+
+export default function Button({ width, margin, children, type, onClick }) {
+  return (
+    <StyledButton type={type} width={width} margin={margin} onClick={onClick}>
+      {children}
+    </StyledButton>
+  );
+}
```

Comment:
Is this component necessary? I guess you can use the StyledButton, because the Button does not contribute any functionality or renaming of props.

Reply:
It's not strictly necessary, no. Seeing as how this is a demo that the students will be confronted with during the lecture part of the session, I figured it'd be best to keep any styling as far away as possible and the code as logic-focused as possible. Does that make sense?

Reply:
Not really, I would teach the students to write their code as simple and understandable as possible, adding an extra component layer that does nothing but renaming the component locally in the file introduces more complexity and more stuff to read/parse/understand. I would argue lets remove this extra component, the StyledButton should be kept in this extra file of course. So you have the separation of styling and functionality but not two components in one file where one is obsolete.

Reply:
Seeing as how this is a demo, the students shouldn't even be shown the content of the styled components. 🤷‍♀️ I'll try to find a solution.